samples/nrf9160/serial_lte_modem/src/main.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <logging/log.h>
+LOG_MODULE_REGISTER(app, CONFIG_SLM_LOG_LEVEL);
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <uart.h>
+#include <string.h>
+#include <bsd.h>
+#include <lte_lc.h>
+
+#include <hal/nrf_gpio.h>
+#include <hal/nrf_power.h>
+#include <hal/nrf_regulators.h>
+
+#include "slm_at_host.h"
+
+#if defined(CONFIG_SLM_CONNECT_UART_0) && defined(CONFIG_SLM_GPIO_WAKEUP)
+#error CONFIG_SLM_GPIO_WAKEUP should not be defined when UART0 is used
+#endif
+
+/* global variable used across different files */
+struct at_param_list m_param_list;
+
+/**@brief Recoverable BSD library error. */
+void bsd_recoverable_error_handler(uint32_t err)
+{
+	LOG_ERR("bsdlib recoverable error: %u", err);
+}
+
+/**@brief Irrecoverable BSD library error. */
+void bsd_irrecoverable_error_handler(uint32_t err)
+{
+	LOG_ERR("bsdlib irrecoverable error: %u", err);
+	__ASSERT_NO_MSG(false);
+}
+
+void start_execute(void)
+{
+	int err;
+
+	LOG_INF("Serial LTE Modem");
+#if defined(CONFIG_SLM_AT_MODE)
+	err = at_host_init();
+	if (err !=0) {
+		LOG_ERR("Failed to init at_host: %d", err);
+		return;
+	}
+
+	/* Initialize AT Parser */
+	err = at_params_list_init(&m_param_list, CONFIG_SLM_AT_MAX_PARAM);
+	if (err) {
+		LOG_ERR("Failed to init AT Parser: %d", err);
+		return;
+	}
+#endif
+}
+
+#ifdef CONFIG_SLM_GPIO_WAKEUP
```

Comment:
@carlescufi Could anyone more familiar with low-power modes in nRF have a look at this part?

Reply:
@pizi-nordic and @anangl could you please take a look at this particular part?